board/ipaccess/ipa73xx/ipa73xx.c

```diff
@@ -129,6 +129,37 @@ DECLARE_GLOBAL_DATA_PTR;
 
 /* Functions --------------------------------------------------------------- */
 
+
+/* Storage of ecc_check_result in SRAM:
+ *   0x2001ffac - ecc_check_result reversed copy 2
+ *   0x2001ffa8 - ecc_check_result copy 2
+ *   0x2001ffa4 - ecc_check_result reversed copy 1
+ *   0x2001ffa0 - ecc_check_result copy 1
+ */
+#define SRAM_PK_ECC_RESULT_ADDRESS 0x2001ffa0
+
+static void load_pk_ecc_result(void)
+{
+    ulong r0, r1, r2, r3;
+    volatile ulong* pk_ecc_result_addr = (volatile ulong*)(SRAM_PK_ECC_RESULT_ADDRESS);
+    
+    r3 = ~(pk_ecc_result_addr[3]);
+    r2 = pk_ecc_result_addr[2];
+    r1 = ~(pk_ecc_result_addr[1]);
+    r0 = pk_ecc_result_addr[0];
+    
+    if ( (r0 == r1) && (r0 == r2) && (r0 == r3) )
+    {
+        char result_str[11];
+        snprintf(result_str, sizeof(result_str), "%d", (int)r0);
```

Comment:
I don't get the point about wrapping - aren't we just talking about 32-bit numbers, whether we interpret them as signed or unsigned? And isn't this code specific to this platform, and therefore doesn't need to be portable? The origin of these numbers is in the spl assembly code, which calls the rs_decode() function, which returns a signed integer number in r0. Zero or positive indicates the number of corrections, -1 indicates uncorrectable errors. The -2 value (and now perhaps -3) just extend this set. I guess there's no reason why we can't express all this as positive integers and specify 0xffffffff as the error case, with 0xfffffffe and 0xfffffffd covering the other cases. Or we could use any other positive numbers we choose, provided they are greater than the maximum number of correctable error symbols (64 if I remember correctly - two separate chunks of 32). Happy to do something like that, but is it really necessary?